src/charm.py

```diff
@@ -68,9 +62,16 @@ def __init__(self, *args):
         self.secrets = SecretCache(self)
         self.status = MongosStatusHandler(self)
 
+        self.node_port_manager = NodePortManager(
+            self, pod_name=self.unit.name.replace("/", "-"), namespace=self.model.name
+        )
+
     # BEGIN: hook functions
     def _on_mongos_pebble_ready(self, event) -> None:
         """Configure MongoDB pebble layer specification."""
+        # any external services must be created before setting of properties
+        self.update_external_services()
```

Comment:
**question:** Where/When do you set your config file? `mongod.conf` or whatever?

Reply:
We do not set a config file, instead we supply command line arguments and then send that to the pebble layer :) 

I could be wrong, but I think you're wanting to know what IP do we configure our service to. To that I answer  `--bind_ip_all`

if you want to see how those are generated [you can see this function](https://github.com/canonical/mongos-k8s-operator/blob/65377bbdf4c7865aa2553b6bb977f84b26bc26dc/lib/charms/mongodb/v1/helpers.py#L106) - note that this link is for a separate branch as its part of different work

Reply:
No no, you answered my follow-up question.
I wanted to confirm that you would be able to set some config-option that depended on the services node-port + node-ip existing, but it's `pebble-ready` and you have run-args, so by that point the nodeport services will be set-up and ready. 👍🏾